experiments/hadoop-yarn/ansible/22-config-spark-master.yml

```diff
@@ -49,20 +50,23 @@
         insertbefore: "EOF"
         block: |
             # https://spark.apache.org/docs/3.0.0-preview2/running-on-yarn.html#spark-properties
-            spark.master            yarn
-            spark.driver.memory              17g
-            spark.yarn.am.memory            17g
-            spark.executor.memory          17g
-            spark.executor.cores            4
-            spark.executor.instances    4
-            spark.yarn.am.cores  4
-            spark.eventLog.enabled  true
-            spark.driver.maxResultSize	8192m
-            spark.local.dir         {{sptemplink}}
-            spark.master            yarn
-            spark.eventLog.enabled  true
-            spark.eventLog.dir      hdfs://{{hdhost}}:9000/spark-log
-    
+            spark.master                yarn
+
+            spark.driver.memory          13g
+            spark.yarn.am.memory         13g
+            spark.yarn.am.cores            4
+
+            spark.executor.memory        13g
+            spark.executor.cores           4
+            spark.executor.instances      11
+
+            spark.eventLog.enabled      true
+            spark.driver.maxResultSize 8192m
+
+            spark.local.dir            {{sptemplink}}
+            spark.eventLog.dir         hdfs://{{hdhost}}:9000/spark-log
+
+
```

Comment:
Do we have a benchmark that shows these optimizations are infact improving performance that we can point to?

Reply:
Wondering if the performance optimization changes might have been better off on a separate branch than the mounts. (Although I realize that they are related and mounts do affect performance)

Reply:
Your tests from `20210211-ML-Notebook-Benchmarking` showed that using local disc was a lot faster, but the tiny and small VMs didn't have enough local disc to cache the whole data set.
The aim of these changes was to update the deployment to use medium VMs, which have a different disc layout, different number of cores and different amount of memory.
The configuration changes were the minimum needed to get it to perform to the same level as the results you reported for local disc in `20210211-ML-Notebook-Benchmarking` but with enough memory and disc space to handle the full data set. 
There is indeed a lot of scope for performance optimization in the future.